Sources/SimpleHTTP/MultipartForm/MultipartFormData.swift

```diff
@@ -48,15 +48,30 @@ enum Boundary {
 
 struct BodyPart {
     let headers: [Header]
-    let stream: InputStream
     let length: Int
+    let stream: () throws -> InputStream
+
     var hasInitialBoundary = false
     var hasFinalBoundary = false
 
-    init(headers: [Header], stream: InputStream, length: Int) {
+    private init(headers: [Header], stream: @escaping () throws -> InputStream, length: Int) {
         self.headers = headers
-        self.stream = stream
         self.length = length
+        self.stream = stream
+    }
+
+    init(headers: [Header], url: URL, length: Int) {
+        let stream = {
```

Comment:
Hmm Unit tests are not working. For now the easiest thing would to explicitly set the type of the closure and i'll bump minimum swift version later. WDYT?

Reply:
ok